crates/typst/src/eval/tracer.rs

```diff
@@ -60,12 +63,30 @@ impl Tracer {
         }
     }
 
-    /// Add a warning.
+    /// Add a warning. If the warning was emitted by a package that has its warnings suppressed, no warning is added.
```

Comment:
@\reviewer I wonder if that breaks memoization/incremental builds because the order of operation is relevant?

It might be better to adjust `Tracer.warnings()` instead, so that it filters all warnings when reading them. Any opinions?